packages/oauth/src/createPublicOauthClient.ts

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import delay from "delay";
+import type {
+  AuthorizationServer,
+  Client,
+  HttpRequestOptions,
+  OAuth2TokenEndpointResponse,
+} from "oauth4webapi";
+import {
+  authorizationCodeGrantRequest,
+  calculatePKCECodeChallenge,
+  customFetch,
+  generateRandomCodeVerifier,
+  generateRandomState,
+  processAuthorizationCodeOAuth2Response,
+  processRevocationResponse,
+  refreshTokenGrantRequest,
+  revocationRequest,
+  validateAuthResponse,
+} from "oauth4webapi";
+import invariant from "tiny-invariant";
+import { TypedEventTarget } from "typescript-event-target";
+import type { Events, PublicOauthClient } from "./PublicOauthClient.js";
+import { throwIfError } from "./throwIfError.js";
+import type { Token } from "./Token.js";
+
+const storageKey = "asdfasdfdhjlkajhgj";
```

Comment:
Is this right? Is this just supposed to be some random string for the key?

Reply:
the value really doesn't matter :) As long as we keep using the same key all is well.